### PR TITLE
Experimental change to remove DescribeAccount call

### PR DESCRIPTION
**Pull request by connelldave**
I think this is a simpler first pass at the problem - untested in anything real, but I can't see it being a particularly big change?

Let me know what you think :) Deprecating the org_master setting is an easy next PR from your work too, I'm not sure there's anything else dependant?

### CODE REVIEW

File: botocove/cove_session.py [hunk 1]
```diff
@@ -44,21 +43,6 @@ def activate_cove_session(self) -> "CoveSession":
             f"{self.session_information['RoleName']}"
         )
 
-        if self.org_master:
-            try:
-                account_description: AccountTypeDef = self.org_client.describe_account(
```
Comment (iainelder) [inline botocove/cove_session.py:49]: I think the org_client could be entirely removed now from the CoveSession class. I don't see it used anywhere. Is there another reason to keep it?

File: tests/test_session.py [hunk 2]
```diff
@@ -1,11 +1,9 @@
-from typing import Any, List
+from typing import List
 
 import pytest
 from boto3 import Session
-from botocore.exceptions import ClientError
 from mypy_boto3_organizations.type_defs import AccountTypeDef
 from mypy_boto3_sts.type_defs import PolicyDescriptorTypeTypeDef
-from pytest_mock import MockerFixture
```
Comment (iainelder) [inline tests/test_session.py:8]: I think we can remove the pytest_mock dependency now. It's not used anywhere else.

File: botocove/cove_host_account.py [hunk 3]
```diff
@@ -229,21 +264,37 @@ def _get_accounts_by_organization_units(
...
-            .paginate()
-            .build_full_result()["Accounts"]
-        )
-        return {acc["Id"] for acc in all_org_accounts if acc["Status"] == "ACTIVE"}
+        """
+        Captures all account metadata into self.account_data for future lookup
+        and returns a set of account IDs in the AWS organization.
+        """
+        pages = self.org_client.get_paginator("list_accounts").paginate()
+        self.account_data: Dict[str, AccountTypeDef] = {
```
Comment (iainelder) [inline botocove/cove_host_account.py:272]: In my earlier attempt I tried to do everything with pure functions. This method has the side effect of setting account_data. That's a neat trick that makes the implementation easier. I think I like it!

**Review by iainelder — APPROVED** [review]
It's a nice focussed change to remove the DescribeAccount call. I've only read the code, not executed it yet, but it looks good to me, so I'll give it my approval.

What I would really like to see now is a chart like those in #20 and #40 to see how dropping DescribeAccount affects the performance. I'm hoping botocove will go faster.

File: botocove/cove_session.py (hunk 1 above)
Comment (connelldave) [reply botocove/cove_session.py:49]: I doubt it, I intentionally avoided any deprecation or non performance clean up in this PR though

**Pull request merged**